src/stan/mcmc/hmc/static_uniform/adapt_diag_e_static_uniform.hpp

```diff
@@ -31,10 +31,15 @@ class adapt_diag_e_static_uniform
       this->stepsize_adaptation_.learn_stepsize(this->nom_epsilon_,
                                                 s.accept_stat());
 
-      bool update = this->var_adaptation_.learn_variance(this->z_.inv_e_metric_,
-                                                         this->z_.q);
+      Eigen::VectorXd inv_metric;
+
+      bool update
+          = this->var_adaptation_.learn_variance(inv_metric, this->z_.q);
       if (update) {
+        this->z_.set_inv_metric(inv_metric);
```

Comment:
[optional] At this point it's the last time you use `inv_metric` so you could just move it like

```suggestion
        this->z_.set_inv_metric(std::move(inv_metric));
```